common/src/main/java/org/apache/nemo/common/ir/IRDAG.java

```diff
@@ -110,8 +109,16 @@ public boolean advanceDAGSnapshot(final BiFunction<IRDAG, IRDAG, Boolean> checke
     return canAdvance;
   }
 
+  /**
+   * @return a IR DAG summary string, consisting of only the original vertices generated from the frontend.
+   */
   public String irDAGSummary() {
-    return "RV" + getRootVertices().size() + "_V" + getVertices().size() + "_E" + getVertices().stream()
+    return "rv" + this.getRootVertices().size()
+      + "_v" + this.getVertices().stream()
+      .filter(v -> !v.getClass().getPackage().getName().contains("vertex.utility"))  // Exclude utility vertices
```

Comment:
why do we exclude `vertex,utility` here? and is it good for the `DAG` to know which vertices should be excluded? 
Maybe we should need a utility function that filters specific vertices instead of modifying this function.

Reply:
Thanks! It's been fixed